riak_test/yz_stat_test.erl

```
@@ -0,0 +1,102 @@
+%% -------------------------------------------------------------------
+%%
+%% Copyright (c) 2013 Basho Technologies, Inc.  All Rights Reserved.
+%%
+%% -------------------------------------------------------------------
+%%
+%% @doc Check that stats run as expected
+-module(yz_stat_test).
+-compile(export_all).
+-import(yz_rt, [host_entries/1,
+                wait_for_joins/1,
+                join_all/1,
+                select_random/1]).
+-include_lib("eunit/include/eunit.hrl").
+
+-define(CFG, [{yokozuna, [{enabled, true}]}]).
+
+confirm() ->
+    random:seed(now()),
+    Cluster = prepare_cluster(1),
+    confirm_stats(Cluster),
+    pass.
+
+prepare_cluster(NumNodes) ->
+    Nodes = rt:deploy_nodes(NumNodes, ?CFG),
+    Cluster = join_all(Nodes),
+    wait_for_joins(Cluster),
+    rt:wait_for_cluster_service(Cluster, yokozuna),
+    Cluster.
+
+create_indexed_bucket(Pid, Cluster, IndexBucket) ->
+    create_indexed_bucket(Pid, Cluster, IndexBucket, IndexBucket).
+
+create_indexed_bucket(Pid, Cluster, Index, Bucket) ->
+    ?assertEqual(ok, riakc_pb_socket:create_search_index(Pid, Index)),
+    ?assertEqual(ok, riakc_pb_socket:set_bucket(Pid, Bucket, [{yz_index, Index}])),
+    yz_rt:wait_for_index(Cluster, binary_to_list(Index)),
+    ok.
+
+%% populate random plain text values
+populate_data(_, _, 0, Acc) -> Acc;
+populate_data(Pid, IB, Count, Acc)-> 
+    KV = gen_random_name(16),
+    PO = riakc_obj:new(IB, KV, KV, "text/plain"),
+    {ok, _Obj} = riakc_pb_socket:put(Pid, PO, [return_head]),
+    populate_data(Pid, IB, Count - 1, [KV|Acc]).
+
+populate_data_and_wait(Pid, Cluster, IndexBucket, Count) ->
+    Values = populate_data(Pid, IndexBucket, Count, []),
+    Search = <<"text:*">>,
+    F = fun(_) ->
+        {ok,{search_results,R,Score,Found}} =
+            riakc_pb_socket:search(Pid, IndexBucket, Search, []),
+        case Found of
```

Using this case here will cause the function to throw a `case_clause` exception instead of returning false.  As shown below.  Instead you want to do something like.

``` erlang
(Count == Found) and (Score =/= 0.0)
```

Here's the error I hit.

```
================ yz_stat_test failure stack trace =====================
{{case_clause,8},
 [{yz_stat_test,'-populate_data_and_wait/4-fun-0-',4,
                [{file,"riak_test/yz_stat_test.erl"},{line,54}]},
  {rt,wait_until,3,[{file,"src/rt.erl"},{line,478}]},
  {yz_rt,'-wait_until/2-fun-0-',3,[{file,"riak_test/yz_rt.erl"},{line,220}]},
  {yz_rt,'-wait_until/2-lc$^0/1-0-',2,
         [{file,"riak_test/yz_rt.erl"},{line,220}]},
  {yz_rt,wait_until,2,[{file,"riak_test/yz_rt.erl"},{line,220}]},
  {yz_stat_test,populate_data_and_wait,4,
                [{file,"riak_test/yz_stat_test.erl"},{line,61}]},
  {yz_stat_test,confirm_stats,1,
                [{file,"riak_test/yz_stat_test.erl"},{line,83}]},
  {yz_stat_test,confirm,0,[{file,"riak_test/yz_stat_test.erl"},{line,21}]}]}
=======================================================================
```